src/java/htsjdk/samtools/util/SequenceUtil.java

```diff
@@ -62,15 +99,27 @@ public static String reverseComplement(final String sequenceData) {
         return htsjdk.samtools.util.StringUtil.bytesToString(bases);
     }
 
-    /** Attempts to efficiently compare two bases stored as bytes for equality. */
-    public static boolean basesEqual(byte lhs, byte rhs) {
-        if (lhs == rhs) return true;
-        else {
-            if (lhs > 90) lhs -= 32;
-            if (rhs > 90) rhs -= 32;
-        }
 
-        return lhs == rhs;
+    /**
+     * Efficiently compare two IUPAC base codes, simply returning true if they are equal (ignoring case),
+     * without considering the set relationships between ambiguous codes.
+     */
+    public static boolean basesEqual(final byte lhs, final byte rhs) {
+        return (bases[lhs] == bases[rhs]);
+    }
+
+    /**
+     * Efficiently compare two IUPAC base codes, one coming from a read sequence and the other coming from
+     * a reference sequence, using the reference code as a 'pattern' that the read base must match.
+     *
+     * We take ambiguous codes into account, returning true if the set of possible bases
+     * represented by the read value is a (non-strict) subset of the possible bases represented
+     * by the reference value.
+     *
+     * Since the comparison is directional, make sure to pass read / ref codes in correct order.
+     */
+    public static boolean basesMatchWithAmbiguity(final byte readBase, final byte refBase) {
+        return (bases[readBase] & bases[refBase]) == bases[readBase];
```

Comment:
Something was bugging me about the asymmetry of this method, and I think it's the name.  `basesMatchWithAmbiguity` sounds like it should be reflexive, but the method is not.  Calling this something like `readBaseCanMatchRefBase` or similar that implies a directionality to the comparison might make this less error prone.